### ISSUE TITLE
System.IO.PathTooLongException

### ISSUE THREAD

**Issue by lukeb1961** [Azure/azure-documentdb-datamigrationtool#56]
1.8.1
.\DT.EXE /s:DocumentDB /s.ConnectionString:" AccountEndpoint=https://MYCOSMOSDBACCOUNT.documents.azure.com:443/;AccountKey=MYCOSMOSDBACCOUNTKEY;Database=MYCOSMOSDBDATABASENAME" /s.Collection:MYCOSMOSDBCOLLECTIONAME /t:JsonFile /t.File:" blobs://DefaultEndpointsProtocol=https;AccountName=MYSTORAGEACCOUNTNAME;AccountKey=MYSTORAGEACCOUNTKEY;EndpointSuffix=core.windows.net@MYSTORAGEACCOUNT.blob.core.windows.net:443/MYCONTAINERNAME/MYBLOBNAME" /t.Overwrite

Transferred: 0; Failed: 0; Elapsed Time: 00:00:04.7850823
Critical error: System.IO.PathTooLongException: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
   at System.IO.PathHelper.GetFullPathName()
   at System.IO.Path.LegacyNormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at Microsoft.DataTransfer.Basics.Files.Sink.LocalFile.LocalFileSinkStreamProvider.CreateStream(CancellationToken cancellation) in F:\_Bld\16189\6357\Sources\_repo\Shared\Microsoft.DataTransfer.Basics.Files\Sink\LocalFile\LocalFileSinkStreamProvider.cs:line 29
   at Microsoft.DataTransfer.JsonFile.Sink.JsonFileSinkAdapterFactory.<CreateAsync>d__2.MoveNext() in F:\_Bld\16189\6357\Sources\_repo\JsonFile\Microsoft.DataTransfer.JsonFile\Sink\JsonFileSinkAdapterFactory.cs:line 36
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.DataTransfer.Core.Service.DataTransferService.<TransferAsync>d__6.MoveNext() in F:\_Bld\16189\6357\Sources\_repo\Core\Microsoft.DataTransfer.Core\Service\DataTransferService.cs:line 70
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.DataTransfer.ConsoleHost.App.Handlers.OneTimeDataTransferHandler.<RunAsync>d__6.MoveNext() in F:\_Bld\16189\6357\Sources\_repo\Console\Microsoft.DataTransfer.ConsoleHost\App\Handlers\OneTimeDataTransferHandler.cs:line 52

DocumentDB Data Migration Tool v1.8.0.0

in other words, this tool flat doesn't work when trying to export a collection to a blob store. Am I missing something?

**Comment by nahk-ivanov** [comment]
Hi,

Looking at [BlobFileStreamProvidersFactoryBase.cs:10](https://github.com/Azure/azure-documentdb-datamigrationtool/blob/af17729809247558ee9c2db2ff86b0882df072d9/Shared/Microsoft.DataTransfer.Basics.Files/Shared/BlobFileStreamProvidersFactoryBase.cs#L10) it matches file with `^blobs?://`. Your command line appears to have a space or other special character in the `/t.File` at the beginning, which prevents regular expression from matching and therefore treat it as a local file.

Regards,
Alex.

**Comment by lukeb1961** [comment]
OK that was part one of my stupidity. The second part was the form for the /t.File path.
The below works fine now. Thank you!

.\DT.EXE 
/ErrorDetails:All 
/s:DocumentDB 
/s.ConnectionString:"AccountEndpoint=https://MYCOSMOSDBACCOUNT.documents.azure.com:443/;AccountKey=MYCOSMOSDBACCOUNTKEY;Database=MYCOSMOSDBDATABASENAME" 
/s.Collection:MYCOSMOSDBCOLLECTIONAME 
/t:JsonFile /t.File:"blobs://MYSTORAGEACCOUNTKEY@MYSTORAGEACCOUNT.blob.core.windows.net:443/MYCONTAINERNAME/MYBLOBNAME"
/t.Overwrite
